type-tests/package.json

```diff
@@ -0,0 +1,7 @@
+{
+  "name": "type-tests",
+  "private": true,
+  "devDependencies": {
+    "ember-modifier": "link:.."
```

Comment:
The approach here treats the `type-tests` directory as its own package with a dependency on `ember-modifier`. This ensures that `typesVersions` will kick in as appropriate and avoids needing a separate `tsconfig` for each TS version we want to test.

Reply:
Oh, that's excellent. I'd been wracking my brains trying to come up with a way to enforce that.